protocol/message.go

```diff
@@ -39,10 +57,39 @@ type DirectoryProof struct {
 	AP    *m.AuthenticationPath
 	STR   *m.SignedTreeRoot
 	TB    *m.TemporaryBinding `json:",omitempty"`
-	Error ErrorCode           `json:",omitempty"`
+	Error ErrorCode
+}
+
+type DirectoryProofInEpoch struct {
+	Type  int
+	AP    *m.AuthenticationPath
+	STR   []*m.SignedTreeRoot
+	Error ErrorCode
+}
+
+type DirectoryProofs struct {
```

Comment:
> I think the client should lookup for range from lookup epoch to its current latest epoch (the epoch of current saved STR).

I think this works since we can assume that the current saved epoch will always be greater than the lookup epoch, because the client should monitor/catch up as soon as the account logs in.

> I originally thought that if the client want to do a key lookup in epoch, say, a month ago, then I'm not sure how big the response's size is?

This is definitely a good point. For such a long request, we should definitely have some way of dealing with a large response size. Maybe `Limit` could be used to indicate the maximum size for a batched response?

> Let me think more about it ... 

Definitely!